### DISCUSSION TITLE
Any interest in adding Swift to the NDK?

### DISCUSSION THREAD

**Discussion by finagolfin** [android/ndk#1745] [Q&A] [answered]
I just saw the Rust roadmap, #1742, and wondered why there couldn't be one for Swift too. [I provide Android SDKs for Swift that work with the official prebuilt Swift compiler for linux from Apple](https://github.com/buttaface/swift-android-sdk), no reason that couldn't be in the NDK since it is entirely open source.

I'm able to cross-compile and run the tests for many Swift packages on Android in my daily CI shown there, and companies like [flowkey](https://medium.com/@ephemer/why-we-put-an-app-in-the-android-play-store-using-swift-96ac87c88dfc) and [Readdle](https://medium.com/android-news/swift-for-android-our-experience-and-tools-7a2f0ba58ab3) have been using Swift in their Android apps for years now. No reason the NDK couldn't enable more app devs to do the same, and I bet there are plenty who'd love to reuse their iOS business logic written in Swift in an Android app.

As a sign of interest, if you are an app dev who'd like Swift in the NDK, just upvote this question.

**Comment by DanAlbert** [comment, marked as answer]
The most important part of the Rust thread is at the top:

> **This is not being worked on, nor is it planned. This bug documents the work that needs to be done if that changes.**

If you want to know the story for Swift, it's basically the same. Copy/paste and `s/rust/swift/i`.

> No reason the NDK couldn't enable more app devs to [use switft]

Staffing. The reason is staffing. IMO we're not adequately staffed to support C++, so definitely not also Rust and Swift.

**Comment by finagolfin** [comment]
@DanAlbert, has this question been shadowbanned? I don't see it in the list of discussions or Q&As.

**Reply by finagolfin** [reply]
Huh? It shows up now that I posted in it.

**Reply by DanAlbert** [reply]
That's not a power we have.

**Comment by finagolfin** [comment]
> That's not a power we have.

Sorry, it's very weird what shows up in the list of discussions: it was showing discussions months older than this one, but not this one. Once I posted another comment though, it strangely showed up again.
